feat: improve inbound connector documentation

chillleader opened the pull request:
## Description

Inbound Connectors documentation improvements:
- Added more details on inbound connector types, updated the existing info about configuring inbound connectors
- Added a note that Operate is required for inbound connectors and an Operate configuration example (Docker)
- Removed the false deprecation warning

<!-- This helps the reviewers by providing an overview of what to expect in the PR. Linking to an issue is preferred but not required. -->
<!-- Add an assignee and use component: labels for good hygiene! -->

## When should this change go live?

<!-- PRs merged go to stage.docs.camunda.io first and must be manually released to docs.camunda.io. -->
<!-- Help the DevEx team prioritize our work (reviews, merges, etc.) by opening PRs sooner. -->

- [ ] This change is not yet live and should not be merged until {ADD_DATE} (apply `hold` label or convert to draft PR)?
- [ ] There is no urgency with this change.
- [ ] This change or page is part of a marketing blog, conference talk, or something else on a schedule.
- [x] This functionality is already available but undocumented.
- [ ] This is a bug fix or security concern.

## PR Checklist

<!-- Keep in mind, Camunda maintains 18 months of versions. Backporting your change or including it in multiple versions is common. -->

- [x] I have added changes to the relevant `/versioned_docs` directory, or they are not for an **already released version**.
- [x] I have added changes to the main `/docs` directory (aka `/next/`), or they are not for **future versions**.
- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

akeller commented:
Should we expect personas working with Connectors to know what stateless and stateful mean?

Rather than define it here, we could add this to the glossary and find a good resource for users needing more info - https://docs.camunda.io/docs/reference/glossary/.

chillleader commented:
@akeller Good point, I removed this wording from the basic docs and added a mention to the glossary.

Still, I would like to give a hint here that inbound connectors are not bound to a specific process instance. As we have noticed, it may be difficult for beginners to grasp this lifecycle difference. I think it may be easier to state this right away :)

christinaausley reviewed (commented):
One comment for linkage 👍

sbuettner commented on docs/components/connectors/connector-types.md:
```diff
@@ -14,13 +14,28 @@ Looking for pre-built, [Out-of-the-box Connectors](/components/connectors/out-of
 ## Inbound Connectors
 
 Inbound Connectors enable workflows to receive data or messages from external systems or services, making it possible to integrate workflows into a wider business process or system architecture.
+Inbound Connectors can be used to trigger a process instance, or to send a message to a running process instance.
```
```suggestion
Inbound Connectors can be used to create a new process instance, or to send a message to a running process instance.
```

sbuettner commented on docs/components/connectors/connector-types.md:
```diff
@@ -14,13 +14,28 @@ Looking for pre-built, [Out-of-the-box Connectors](/components/connectors/out-of
 Inbound Connectors enable workflows to receive data or messages from external systems or services, making it possible to integrate workflows into a wider business process or system architecture.
+Inbound Connectors can be used to trigger a process instance, or to send a message to a running process instance.
+
+The Java code of the inbound Connector has a lifecycle suitable for long-running operations, such as listening for messages on a queue or waiting for a webhook to be called.
```
```suggestion
The Java code of the inbound Connector get `activated` as soon as the Connector Runtime detects an element in a process definition that references an inbound connector. The lifecycle suitable for long-running operations, such as listening for messages on a queue or waiting for a webhook to be called. The code gets `deactivated` in case of an updated or deleted process definition.
```

sbuettner commented on docs/components/connectors/connector-types.md:
```diff
@@ -14,13 +14,28 @@ Looking for pre-built, [Out-of-the-box Connectors](/components/connectors/out-of
+
+The Java code of the inbound Connector has a lifecycle suitable for long-running operations, such as listening for messages on a queue or waiting for a webhook to be called.
+
+Inbound Connector instances are not linked to specific process instances. If the process contains an inbound Connector, the Connector code will be first executed when the process is deployed.
```
```suggestion
Inbound Connector instances are linked to process definitions and not to specific process instances. If a process definition contains an element referencing an inbound Connector, the Connector code will be first executed when the process definition is deployed and the deployment has been detected by the Connector Runtime.
```

akeller commented on docs/components/connectors/connector-types.md:
```diff
@@ -14,13 +14,29 @@ Looking for pre-built, [Out-of-the-box Connectors](/components/connectors/out-of
 
 Outbound Connectors allow workflows to trigger external systems or services, making it possible to integrate workflows with other parts of a business process or system architecture.
 
+The programming code in Java to connect to the external system is executed when the workflow reaches the service task.
```
This sentence is a little awkward. Is there any reason we can't refer to this as "The Java code" like we do a few lines below?

akeller replied:
```suggestion
The Java code to connect to the external system is executed when the workflow reaches the service task.
```

chillleader replied:
I believe it's taken verbatim from [here](https://github.com/camunda/camunda-platform-docs/blob/b75513e4c095bc62dc54fe0291edab4a2f79c985/docs/components/connectors/introduction.md?plain=1#L24). Corrected it here, doesn't look good to me either 🙂

akeller replied:
Always room for improvement 😆 thank you!

akeller reviewed (approved):
I'm good with this.
